### PR TITLE
Fix (core/function): Fix learned round when padding is applied to weights

### PR DESCRIPTION
**Pull request by nickfraser**
Merge #1253 first & rebase. This issue fixes an issue with learned round when it is also combined with a groupwise quantizer when the underlying tensor needs to be padded.

TODO:
 - [x] Use `input_view_impl` when using `learned_round`
 - [x] Add MX + padding test
 - [x] Rebase on #1253

### CODE REVIEW

**Comment by Giuseppe5** [comment]
This is not a draft anymore?

File: src/brevitas/core/function_wrapper/learned_round.py [hunk 1]
```diff
@@ -96,17 +96,20 @@ def __init__(
...
         self.tensor_slicer = SliceTensor()
         self.value = torch.nn.Parameter(learned_round_init)
+        self.input_view_impl = input_view_impl
 
     @brevitas.jit.script_method
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         p = self.learned_round_impl(self.value)
+        p = self.input_view_impl(p)
         p = self.tensor_slicer(p)
         p = (p.to(x.dtype)).view_as(x)
```
Comment (nickfraser) [inline src/brevitas/core/function_wrapper/learned_round.py:114]: Remove this line (should be no-op)

**Review by Giuseppe5 — APPROVED** [review]
